packages/dds/merge-tree/README.md

```diff
@@ -22,6 +22,24 @@ If using any of Fluid Framework's unstable APIs (for example, its `beta` APIs),
 
 <!-- AUTO-GENERATED-CONTENT:END -->
 
+<!-- AUTO-GENERATED-CONTENT:START (README_IMPORT_INSTRUCTIONS:includeHeading=TRUE) -->
+
+<!-- prettier-ignore-start -->
+<!-- NOTE: This section is automatically generated using @fluid-tools/markdown-magic. Do not update these generated contents directly. -->
+
+## Importing from this package
+
+This package leverages [package.json exports](https://nodejs.org/api/packages.html#exports) to separate its APIs by support level.
+For more information on the related support guarantees, see [API Support Levels](https://fluidframework.com/docs/build/releases-and-apitags/#api-support-levels).
+
+To access the `public` ([SemVer](https://semver.org/)) APIs, import via `@fluidframework/merge-tree` like normal.
```

Comment:
I know this is a template but just realized it feels weird to suggest this for packages that have no public exports, only legacy. Should we maybe remove the "." export subpath for these packages and update the markdown magic logic to skip this line if the subpath isn't exported?

Reply:
Omitting the "public" note for packages without a "." exports would be easy to do, but I don't know that we would want to omit such exports altogether. The nice thing with the trimming is that exports update automatically. I'm don't think we want devs to have to worry about adding / removing default package exports when they make API changes.

@jason-ha thoughts?

Reply:
True for the package.json, but can we detect their emptiness in another way? Because importing from public for this package gives you literally nothing:

![image](https://github.com/microsoft/FluidFramework/assets/716334/dcbba96a-a908-4bdb-889b-d1a3b5a04cfc)

Reply:
Potentially, but I think it would require a more thought-out design. In order to actually look at the exports, we would have to build the code before building the docs, which would require some fluid-build dependency changes, among other things.

It also might be nice to not generate the trimmed roll-ups, nor the corresponding API reports when the export set would be empty.

But I'm not sure it's the end of the world to call out expectations around our public APIs for a package without any public exports. In the case where the public exports set is empty, it might be nice to leave a disclaimer that the package has no such exports, but I don't think it's critical.

Reply:
We already require product build to generate doc metadata. (I'm not sure what you are saying Josh about docs without code build.)
Updating task dependencies is pretty easy, if we need more.

I explicitly had to go into flub generate entrypoints and add support for generating empty "roll ups" it is a special case. So, we already know when it is happening in tooling.

I could see an argument for keeping even an empty public export while we were reorganizing, which is why I supported existing cases, but I don't know about the utility now for customers. If it is easier for generally tooling to be allowed to assume there is always a public roll-up file, then I think it is cool to keep them around and doesn't do real harm to customers. (It is kind of nice to be able to see there is a package being resolved even if it doesn't have anything.) In the off chance that we had things not using typescript, then the "public" exports give access to everything. I don't *think* we care about non-type imports, but I've seen some funny stuff out there.

I am not sure what any of the docs logic would do or not do without a public "roll up" or *.public.api.md. The metadata doesn't do anything related to it as far as I know.

Reply:
I also expect that you can strip the "." export and the flub tools would be okay. You'd have to strip the current lint tasks for such packages and their api-extractor config files. (Just saying that I think it would work; not recommending one way or the other.)